src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.cs

```diff
@@ -0,0 +1,1159 @@
+
+    // These tests just validate basic correctness of results in different scenarios, without actually validating
+    // that the incremental nature of this provider works properly.
+
+    [Fact]
+    public void FindCorrectAttributeOnTopLevelClass_WhenSearchingForClassDeclaration1()
+    {
+        var source = @"
+[N1.X]
+class C1 { }
+[N2.X]
+class C2 { }
+
+namespace N1
+{
+    class XAttribute : System.Attribute { }
+}
+
+namespace N2
+{
+    class XAttribute : System.Attribute { }
+}
+";
+        var parseOptions = TestOptions.RegularPreview;
+        Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+
+        Assert.Single(compilation.SyntaxTrees);
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var input = ctx.ForAttributeWithMetadataName<ClassDeclarationSyntax>("N1.XAttribute");
+            ctx.RegisterSourceOutput(input, (spc, node) => { });
+        }));
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+        driver = driver.RunGenerators(compilation);
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.Collection(runResult.TrackedSteps["result_ForAttributeWithMetadataName"],
+            step => Assert.True(step.Outputs.Single().Value is ClassDeclarationSyntax { Identifier.ValueText: "C1" }));
+    }
+
+    [Fact]
+    public void DoNotFindAttributeOnTopLevelClass_WhenSearchingSimpleName1()
+    {
+        var source = @"
+[N1.X]
+class C1 { }
+[N2.X]
+class C2 { }
+
+namespace N1
+{
+    class XAttribute : System.Attribute { }
+}
+
+namespace N2
+{
+    class XAttribute : System.Attribute { }
+}
+";
+        var parseOptions = TestOptions.RegularPreview;
+        Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+
+        Assert.Single(compilation.SyntaxTrees);
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var input = ctx.ForAttributeWithMetadataName<ClassDeclarationSyntax>("XAttribute");
+            ctx.RegisterSourceOutput(input, (spc, node) => { });
+        }));
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+        driver = driver.RunGenerators(compilation);
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.False(runResult.TrackedSteps.ContainsKey("result_ForAttributeWithMetadataName"));
+    }
+
+    [Fact]
+    public void DoNotFindAttributeOnTopLevelClass_WhenSearchingSimpleName2()
```

Comment:
Could combine with the previous test with a `[Theory]`

```
[Theory]
[InlineData("X")]
[InlineData("XAttribute")]
[InlineData("N1.xAttribute")]
public void DoNotFindAttributeOnTopLevelClass_WhenSearchingSimpleName(string name)
```

Reply:
sounds good.  will do.

Reply:
Done!